v7labs/darwin-py
# Sphinx

Why
----

Adding a tool to automatically generate documentation for darwin-py, so our customer can have code documentation for the SDK.

How
----

Using Sphinx and Github Actions to generate new docs every time a new push is made.

How can I test this?
-----

The Github Action `generate-docs` must complete. If that success, a new folder `docs` should appear. You can open that new folder and check for the `index.html` file and open it in your browser. However, this will only happen once the Github Action gets to `master`.

You can still test this locally by downloading the branch and running:
```
# Install deps
python -m pip install --upgrade pip
pip install pytest
pip install pytest-describe
pip install --upgrade setuptools
pip install --editable ".[test]"
pip install torch torchvision
pip install -U sphinx
pip install m2r
pip install sphinx_rtd_theme

# Parse README
rm README.rst
m2r README.md
mv README.rst source/

# Generate docs
rm -rf docs/* 
sphinx-apidoc -f -o source darwin   
sphinx-build -b html source/ docs/
```

Its a lot of dependencies and commands to run, I know, but you can pretty much make a huge one using `&&`:

```
# I assume you have all deps
rm README.rst && m2r README.md && mv README.rst source/ && rm -rf docs/*  && sphinx-apidoc -f -o source darwin && sphinx-build -b html source/ docs/
```

This should create a `docs` folder in the project where you will be able to see the `index.html` file and open it as an entry point.
